src/pyramid/config/security.py

```diff
@@ -21,6 +23,38 @@ class SecurityConfiguratorMixin(object):
     def add_default_security(self):
         self.set_csrf_storage_policy(LegacySessionCSRFStoragePolicy())
 
+    @action_method
+    def set_security_policy(self, policy):
+        """ Override the :app:`Pyramid` :term:`security policy` in the current
+        configuration.  The ``policy`` argument must be an instance
+        of a security policy or a :term:`dotted Python name`
+        that points at an instance of a security policy.
+
+        .. note::
+
+           Using the ``security_policy`` argument to the
+           :class:`pyramid.config.Configurator` constructor can be used to
+           achieve the same purpose.
+
+        """
+
+        def register():
+            self._set_security_policy(policy)
+
+        intr = self.introspectable(
+            'security policy',
+            None,
+            self.object_description(policy),
+            'security policy',
+        )
+        intr['policy'] = policy
+        # authentication policy used by view config (phase 3)
+        self.action(ISecurityPolicy, register, introspectables=(intr,))
+
+    def _set_security_policy(self, policy):
+        policy = self.maybe_dotted(policy)
+        self.registry.registerUtility(policy, ISecurityPolicy)
```

Comment:
I'm not sure what the point of this pattern is but the code should just be done in the register function. Not by adding a hidden method to the configurator. I assume it is a hook for tests but you aren't using it so it shouldn't be done.

Reply:
Roger that.

Question:  Shouldn't policy be resolved (`policy = self.maybe_dotted(policy)`) in the method body, rather than in the config action?  That way the introspectable always gets the actual policy class.  Hesitant to change it since I assume I'm missing something.

Reply:
Yeah in other actions that's done right away at the top of the method.

Reply:
Thanks!  Made the requested changes to the configurator.